Classes/Service/LanguageService.php

```diff
@@ -56,74 +43,43 @@ public function getSourceLanguage(Site $currentSite): array
     }
 
     /**
-     * @return array{uid: int, title: string, language_isocode: string, languageCode: string, formality: string}
+     * @return array{uid: int, title: string, language_isocode: string, languageCode: string, formality: string}|false
      * @throws LanguageRecordNotFoundException
      * @throws LanguageIsoCodeNotFoundException
+     * @throws InvalidArgumentException
      */
-    public function getTargetLanguage(Site $currentSite, int $languageId): array
+    public function getTargetLanguage(Site $currentSite, int $languageId): array|false
```

Comment:
```suggestion
    public function getTargetLanguage(Site $currentSite, int $languageId): ?array
```

Better use nullable return type instead of introducion onion type here. Using type/bool is a oldschool php style dating back to non-typed area. Method code
should also be adopted to return `null` instead of `false`. Calling code should
also be adopted.

**EDIT**

Thinking more about that, also including calling code, I'd say we should not change the return value and instead ensuring to throw a exception in all invalid cases and not mixing exception and null(/false) return values to have a clean way. 

That way calling code does not need to use try-catch AND evaulate the return value to determine if it was successfull or not. (one way instead of two ways which may be missed).

What do you think ?